### PR TITLE
prefix variable consistency across modules

### PR DESCRIPTION
**Pull request by skalolazka**
Modules/ folder ready - consistent usage of the "prefix" variable

Fixes https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/issues/954

### CODE REVIEW

File: modules/cloud-function/variables.tf [hunk 1]
```diff
@@ -94,6 +94,10 @@ variable "prefix" {
   description = "Optional prefix used for resource names."
   type        = string
   default     = null
+  validation {
+    condition     = var.prefix != ""
+    error_message = "Prefix can not be empty, please use null instead."
```
Comment (juliocc) [inline modules/cloud-function/variables.tf:99]: nit: s/can not/cannot/

**Review by juliocc — COMMENTED** [review]
There's also a few cases inside the blueprints folder. What do we want to do with those?

**Comment by skalolazka** [comment]
> 

I'll try to see how long it might take me to go through the blueprints=) Being a perfectionist, I would ideally fix everywhere%)

The other question is, should we add the same in fast/ (adding to existing validation and things)? Example: 00-bootstrap. No default null, can be an empty string though. Right? Wouldn't that break some things? Or am I misreading it?

**Comment by juliocc** [comment]
> > 
> 
> I'll try to see how long it might take me to go through the blueprints=) Being a perfectionist, I would ideally fix everywhere%)
> 
> The other question is, should we add the same in fast/ (adding to existing validation and things)? Example: 00-bootstrap. No default null, can be an empty string though. Right? Wouldn't that break some things? Or am I misreading it?

FAST does its own thing with prefixes. I'd focus only on blueprints for now.

**Review by ludoo — APPROVED** [review]

**Comment by ludoo** [comment]
Thanks :)

**Pull request merged**